src/apple/system.rs

```diff
@@ -450,9 +450,25 @@ impl SystemExt for System {
         get_system_info(libc::KERN_HOSTNAME, None)
     }
 
-    fn get_version(&self) -> Option<String> {
+    fn get_kernel_version(&self) -> Option<String> {
         get_system_info(libc::KERN_OSRELEASE, None)
     }
+
+    fn get_os_version(&self) -> Option<String> {
+        unsafe {
+            // get the size for the buffer first
+            let mut size = 0;
+            get_sys_value_by_name(b"kern.osproductversion\0", &mut size, std::ptr::null_mut());
```

Comment:
Two things to add here: first, please check what `get_sys_value_by_name` returns (if `false`, return `None`). Second, please check that `size > 0`, otherwise return `None`.

Reply:
fixed in 1aff6db